critiquebrainz/db/review.py

```diff
@@ -0,0 +1,625 @@
+import sqlalchemy
+from critiquebrainz import db
+from critiquebrainz.db import exceptions as db_exceptions
+from flask import abort
+from brainzutils import cache
+from werkzeug.exceptions import BadRequest
+from random import shuffle
+from flask_babel import lazy_gettext
+import critiquebrainz.db.revision as db_revision
+from critiquebrainz.db.user import User
+import critiquebrainz.db.users as db_users
+from critiquebrainz.db.user import User
+import uuid
+from datetime import datetime, timedelta
+
+REVIEW_CACHE_NAMESPACE = "Review"
+DEFAULT_LICENSE_ID = "CC BY-SA 3.0"
+DEFAULT_LANG = "en"
+ENTITY_TYPES = [
+    "event",
+    "place",
+    "release_group",
+]
+
+def get_or_404(review_id):
+    """Get a review using review ID or abort with error 404."""
+    try:
+        review = get_by_id(review_id)
+    except db_exceptions.NoDataFoundException:
+        abort(404)
+    return review
+
+
+def to_dict(review, confidential=False):
+    review["user"] = User(db_users.get_by_id(review.pop("user_id")))
```

Comment:
In `review_credit` macro present in `macros.html`, `review.user.avatar` is used to show user avatar which is a property in the `User` class. The macro is used in `popular reviews` section and  `browse reviews` pages.